static/panes/tool.ts

```diff
@@ -0,0 +1,682 @@
+    private readonly editorId?: number;
+    private compilerService: any;
+    private readonly monacoStdin: string | null;
+    private readonly treeId: number;
+    private codeLensProvider?: monaco.languages.CodeLensList;
+
+    constructor(hub: Hub, container: Container, state: MonacoPaneState & ToolState) {
+        super(hub, container, state);
+        this.hub = hub;
+        this.compilerId = state.compiler;
+        this.editorId = state.editorid;
+        this.treeId = state.tree;
+        this.toolId = state.toolId;
+        this.toolName = 'Tool';
+        this.compilerService = hub.compilerService;
+        this.editorContentRoot = this.domRoot.find('.monaco-placeholder');
+        this.plainContentRoot = this.domRoot.find('pre.content');
+        this.optionsToolbar = this.domRoot.find('.options-toolbar');
+        this.badLangToolbar = this.domRoot.find('.bad-lang');
+        this.compilerName = '';
+        this.monacoStdin = state.monacoStdin || null;
+        this.monacoEditorOpen = state.monacoEditorOpen || false;
+        this.monacoEditorHasBeenAutoOpened = state.monacoEditorHasBeenAutoOpened || false;
+        this.monacoStdinField = '';
+        this.normalAnsiToHtml = makeAnsiToHtml();
+
+        this.optionsField = this.domRoot.find('input.options');
+        this.localStdinField = this.domRoot.find('textarea.tool-stdin');
+
+        this.fontScale = new FontScale(this.domRoot, state, '.content');
+        this.fontScale.on('change', () => {
+            this.saveState();
+        });
+
+        this.createToolInputView = () => {
+            return Components.getToolInputViewWith(this.compilerId + '', this.toolId + '', this.toolName ?? '');
+        };
+
+        this.initButtons(state);
+        this.options = new Toggles(this.domRoot.find('.options'), state as unknown as Record<string, boolean>);
+        this.options.on('change', _.bind(this.onOptionsChange, this));
+
+        this.paneRenaming = new PaneRenaming(this, state);
+
+        this.initArgs(state);
+        this.initCallbacks();
+
+        this.onOptionsChange();
+
+        this.eventHub.emit('toolOpened', this.compilerId, this.currentState());
+        this.eventHub.emit('requestSettings');
+    }
+
+    override getInitialHTML(): string {
+        return $('#tool-output').html();
+    }
+
+    override registerOpeningAnalyticsEvent() {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'Tool',
+        });
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            monacoConfig.extendConfig({
+                readOnly: true,
+                language: 'text',
+                fontFamily: 'courier new',
+                lineNumbersMinChars: 5,
+                guides: undefined,
+            })
+        );
+    }
+
+    initCallbacks() {
```

Comment:
https://github.com/compiler-explorer/compiler-explorer/blob/main/static/panes/pane.ts#L174 already implements some of the binds here :)